### ISSUE TITLE
Failed to set ECDH curve

### ISSUE THREAD

**Issue by isamauny** [opticdev/optic#1608]
Hi there,

I am trying to use Optic to capture traffic to a TLS endpoint. I did setup the TLS cli certificate, but still hit the $subject issue. 

My target endpoint uses the following ciphers:

# TLS 1.3 (suites in server-preferred order)
TLS_AES_256_GCM_SHA384 (0x1302)   ECDH x25519 (eq. 3072 bits RSA)   FS	256
TLS_CHACHA20_POLY1305_SHA256 (0x1303)   ECDH x25519 (eq. 3072 bits RSA)   FS	256
TLS_AES_128_GCM_SHA256 (0x1301)   ECDH x25519 (eq. 3072 bits RSA)   FS	128
# TLS 1.2 (suites in server-preferred order)
TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 (0xc02f)   ECDH x25519 (eq. 3072 bits RSA)   FS	128
TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 (0xc030)   ECDH x25519 (eq. 3072 bits RSA)   FS	256
TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256 (0xcca8)   ECDH x25519 (eq. 3072 bits RSA)   FS	256

Should this be working or have I hit a limitation of the embedded proxy ? 
BTW, I have tried proxying to same API with a 1.3 MITM proxy and it works.

Thanks (and hi @acunniffe !)

**Comment by acunniffe** [comment]
Oh hi @isamauny ! We'll look into this more today. 

In the meantime if you did manage to get a HAR out of MITM proxy, did you know you can pipe that directly into Optic? 

```
optic oas capture [openapi] [target-host] --har /path/to/file.har
```

That should unblock you as we look into this

**Comment by acunniffe** [comment]
@isamauny 

Can you share the openssl version on your local machine? And also the API server you are targeting? 

If you have time, could you also try targetting another API host to see if it's service specific. For instance `https://api.github.com`?

**Comment by Manouchehri** [comment]
> Oh hi @isamauny ! We'll look into this more today.
> 
> In the meantime if you did manage to get a HAR out of MITM proxy, did you know you can pipe that directly into Optic?
> 
> ```
> optic oas capture [openapi] [target-host] --har /path/to/file.har
> ```
> 
> That should unblock you as we look into this

Is it still okay to use `oas`? I see it's marked as deprecated.

**Comment by niclim** [comment]
> Is it still okay to use oas? I see it's marked as deprecated.

@Manouchehri - all `oas` commands are now under the root command (i.e. `optic capture` instead of `optic oas capture`, etc). Commands run identically, we're just removing the `oas` subcommand here for simplicity

**Comment by Manouchehri** [comment]
Hmm, I still run into this issue with newer versions of node, might want to leave the ticket open until more troubleshooting is done. 🙂

**Comment by acunniffe** [comment]
Didn't realize this. Will give this a run again this week and see what we find.

**Comment by niclim** [comment]
Hi, we released a fix in `0.42.16` which fixes this - let me know if this looks good and I can close this out

This appears to have affected version of `node >=17` where the node ssl version was < 3
